### ISSUE TITLE
`panic` lint does not report `panic_any`

### ISSUE THREAD

**Issue by stoneman258** [rust-lang/rust-clippy#13292]
### Summary

I've encountered an issue with the linting for `panic` in my codebase. The lint does not report warnings when `panic_any` used, but it can provide a warning when `panic!` used.

This inconsistency can lead to potential issues. It would be beneficial for the linting to cover both `panic!` and `panic_any` to ensure that all uses of `panics` are appropriately flagged. Besides, some staic analyzers (e.g,. lockbud) can support panic_any detection.

### Lint Name

clippy::panic

### Reproducer

I use Clippy detect the following code:

```rust
pub fn assert_capacity(capacity: usize) {
    if capacity != 0 {
        panic!("Unable to allocate requested capacity") //should report warning
    }
}

pub fn _assert_capacity(capacity: usize) {
    if capacity != 0 {
        std::panic::panic_any("Unable to allocate requested capacity")//should report a warning
    }
}
```
I tried this command:
```
cargo clippy -- -A clippy::all -W clippy::panic
```
I expected to see this happen:
Both `panic!` and `panic_any` should be reported.
Instead, this happened:
Only `panic!` is reported.

### Version

```text
rustc 1.81.0-nightly (4bc39f028 2024-06-26)
binary: rustc
commit-hash: 4bc39f028d14c24b04dd17dc425432c6ec354536
commit-date: 2024-06-26
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```
